tools/toolchains/__init__.py

```diff
@@ -625,7 +625,8 @@ def link_program(self, r, tmp_path, name):
         lib_dirs = r.get_file_paths(FileType.LIB_DIR)
         libraries = [l for l in r.get_file_paths(FileType.LIB)
                      if l.endswith(self.LIBRARY_EXT)]
-        dependencies = objects + libraries + [linker_script] + config_file
+        hex_files = r.get_file_paths(FileType.HEX)
+        dependencies = objects + libraries + [linker_script] + config_file + hex_files
```

Comment:
This was probably quite helpful for development, but won't really change anything for a user. This is fine to stay in, as there's not really a downside that I can think of.

Reply:
I would defend this (and only this one), as in the dual-core application case, event with manual handling, e.g. running:
`$ mbed compile -m FUTURE_SEQUANA_M0 && cp BUILD/ FUTURE_SEQUANA/GCC_ARM/application.hex ./m0_images && mbed compile -m FUTURE_SEQUANA
`
this will prevent the necessity to manually delete previous build result. OK?

Reply:
I was not attacking this one, in fact I like it. It's fine to come in. Sorry that I put it in a request changes review.